common/lib/xmodule/xmodule/modulestore/tests/test_publish.py

```diff
@@ -139,3 +159,208 @@ def test_publish_draft_delete(self):
+                    with test_modulestore.branch_setting(ModuleStoreEnum.Branch.draft_preferred):
+                        test_item = test_modulestore.get_item(self.course_key.make_usage_key(block_type=self.block_type, block_id=self.block_id))
+
+                    # Perform an elemental operation on a particular item.
+                    assert_true(hasattr(test_modulestore, self.operation))
+                    elemental_op = getattr(test_modulestore, self.operation)
+                    op_params = []
+                    for p in self.operation_params:
+                        if isinstance(p, basestring):
+                            if p == '@location':
+                                op_params.append(test_item.location)
+                            elif p == '@user_id':
+                                op_params.append(self.user_id)
+                            else:
+                                op_params.append(p)
+                        else:
+                            op_params.append(p)
+                    elemental_op(*op_params)
+
+                    # Export the course again.
+                    export_course_to_xml(
+                        test_modulestore,
+                        test_content,
+                        self.course_key,
+                        export_dir,
+                        self.EXPORTED_COURSE_AFTER_DIR_NAME,
+                    )
+                    # Verify the export OLX format.
+                    self.check_olx_format(export_dir, self.EXPORTED_COURSE_AFTER_DIR_NAME, self.olx_loc_and_fmt_after)
+
+
+class TestPublishSingleUnit(UniversalTestProcedure):
+    __test__ = False
+    block_type = 'html'
+    block_id = 'unit_0'
+    operation = 'publish'
+    operation_params = ('@location', '@user_id')
+    olx_loc_and_fmt_before = (
+        # Used to check the export before the operation takes place.
+        # List of:
+        #   - path to OLX to check
+        #   - matching OLX
+        (
+            ('drafts', 'html', 'unit_0.xml'),
+            '''
+            <html filename="html_unit"/>
+            '''
+        ),
+        (
+            ('html', 'unit_0.xml'),
+            None,
+        ),
+    )
+    olx_loc_and_fmt_after = (
+        (
+            ('drafts', 'html', 'unit_0.xml'),
+            None, # None means the file/section shouldn't exist.
+        ),
+        (
+            ('html', 'unit_0.xml'),
+            '''
+            <html filename="html_unit"/>
+            '''
+        ),
+    )
+
+
+for modulestore_backend in (
+    MixedMongoModulestoreTest,
+    MixedSplitModulestoreTest,
+    DirectMongoModulestoreTest,
+):
+    for base_test_case in (
+            TestPublishSingleUnit,
+    ):
+
+        test_name = base_test_case.__name__ + "With" + modulestore_backend.__name__
+        test_classes = (base_test_case, modulestore_backend)
+        vars()[test_name] = type(test_name, test_classes, {'__test__': True})
```

Comment:
Given that you only have a single test function, I'd lean towards just using a `ddt` test, rather than this complicated test mixing procedure.